docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+---
+title: on-prem-data-collection
+authors:
+  - "@CrystalChun"
+creation-date: 2022-09-30
+last-updated: 2022-10-28
+---
+
+# On-premises Data Collection
+
+## Summary
+
+The Assisted Installer ("AI") will provide a process to collect data about installations when it is deployed on-premises as part of either [Red Hat Advanced Cluster Management for Kubernetes](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6) ("RHACM") or as a stand-alone Podman deployment. Data was already collected from clusters deployed on [Red Hat's Hybrid Cloud Console](https://console.redhat.com/openshift/assisted-installer/clusters) (“SaaS”), but there was no method for obtaining data from on-premises deployments.
+
+## Motivation
+
+Data provides insight in order to improve the product for our users.
+
+### Goals
+
+1. Create a process to send [event](https://github.com/openshift/assisted-service/tree/master/docs/dev/events.md) data from on-premises deployments of AI.
+2. Analyze and process on-premises data (by use of visualizations, dashboards, etc.) at Red Hat similarly to SaaS data.
+3. Analyze on-premises data both separately and together with SaaS data.
+
+
+### Non-Goals
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+### Data Sending Flow
+
+All of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Data will be sent when:
+1. A cluster installation attempt finishes (regardless of success or failure)
+
+Before the data is sent, Assisted Service will get a remote configuration from an endpoint we specify.
```

Comment:
What do you mean by "get" here?
Isn't this something that would be baked into the deployment/operator?

Reply:
That's right! Decided during meeting that remote configuration would not be part of the first iteration. Thanks!